virttest/qemu_devices/qcontainer.py

```diff
@@ -942,6 +942,26 @@ def get_aavmf_vars(params):
                                                   parent_bus={'aobject': 'pci.0'}))
             return devices
 
+        def machine_s390(cmd=False):
```

Comment:
Related to it I think you should be using `s390x` as name to be consistent with image names (at least Fedora is built for s390x, qemu is called `qemu-system-s390x`).

On the other hand the machine type itself is `s390-*`:

```
$ qemu-system-s390x -M help
Supported machines are:
none                 empty machine
s390-ccw-virtio-2.4  VirtIO-ccw based S390 machine v2.4
s390-ccw-virtio-2.5  VirtIO-ccw based S390 machine v2.5
s390-ccw-virtio-2.6  VirtIO-ccw based S390 machine v2.6
s390-ccw-virtio      VirtIO-ccw based S390 machine v2.7 (alias of s390-ccw-virtio-2.7)
s390-ccw-virtio-2.7  VirtIO-ccw based S390 machine v2.7 (default)
```

So take this only as question/suggestion.